src/optimizer/filter_push_down_optimizer.cpp

```diff
@@ -46,7 +49,16 @@ std::shared_ptr<planner::LogicalOperator> FilterPushDownOptimizer::visitOperator
 std::shared_ptr<LogicalOperator> FilterPushDownOptimizer::visitFilterReplace(
     const std::shared_ptr<LogicalOperator>& op) {
     auto filter = (LogicalFilter*)op.get();
-    predicateSet->addPredicate(filter->getPredicate());
+    auto predicate = filter->getPredicate();
+    if (predicate->expressionType == ExpressionType::LITERAL) {
+        // Avoid executing child plan is literal is Null or False.
```

Comment:
// Avoid executing child plan is literal is Null or False.
->

        // Avoid executing child plan **if** literal is Null or False.